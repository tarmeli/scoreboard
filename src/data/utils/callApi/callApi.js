const callApi = async (route, method) => {
  try {
    const response = await fetch(`http://localhost:4000${route}`, {
      method,
    });

    const {
      data,
      error: { isError, message },
    } = await response.json();

    if (isError) {
      return { callWasSuccessful: false, error: { isError, message } };
    }

    return {
      callWasSuccessful: true,
      data,
      error: {
        isError: false,
        message: "",
      },
    };
  } catch (error) {
    return { callWasSuccessful: false, error: { isError: true, error } };
  }
};

export default callApi;
