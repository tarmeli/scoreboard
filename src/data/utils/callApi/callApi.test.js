import callApi from "./callApi";

describe("callApi", () => {
    const realFetch = global.fetch;

    afterEach(() => {
        global.fetch = realFetch;
    });

    it("given fetch returns data, when called, returns fetched data", async () => {
        global.fetch = () =>
            Promise.resolve({
                json: () => Promise.resolve({ data: "Some data" }),
            });

        const actual = await callApi().then((response) => {
            return response;
        });

        expect(actual).toMatchSnapshot();
    });

    it("given fetch returns error, when called, returns error", async () => {
        global.fetch = () =>
            Promise.resolve({
                json: () => Promise.reject("Some error message"),
            });

        const actual = await callApi().then((response) => {
            return response;
        });

        expect(actual).toMatchSnapshot();
    });
});
