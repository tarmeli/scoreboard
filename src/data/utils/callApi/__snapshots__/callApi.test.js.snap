// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`callApi given fetch returns data, when called, returns fetched data 1`] = `
Object {
  "callWasSuccessful": true,
  "data": "Some data",
  "error": Object {
    "isError": false,
    "message": "",
  },
}
`;

exports[`callApi given fetch returns error, when called, returns error 1`] = `
Object {
  "callWasSuccessful": false,
  "error": Object {
    "error": "Some error message",
    "isError": true,
  },
}
`;
